tt_eager/tt_dnn/op_library/rotary_embedding/multi_core/rotary_embedding_llama_op_multi_core.cpp

```diff
@@ -0,0 +1,304 @@
+        (std::uint32_t)input_cb_index,
+        (std::uint32_t)cos_cb_index,
+        (std::uint32_t)sin_cb_index,
+        (std::uint32_t)trans_mat_cb_index,
+        (std::uint32_t)src_is_dram,
+        (std::uint32_t)cos_is_dram,
+        (std::uint32_t)sin_is_dram,
+        (std::uint32_t)trans_mat_is_dram,
+        (std::uint32_t)Ht,
+        (std::uint32_t)Wt,
+        (std::uint32_t)HtWt,
+        (std::uint32_t)num_rows_per_core,
+        (std::uint32_t)num_sin_cos_rows_per_core
+    };
+    bool dst_is_dram = dst_buffer->buffer_type() == tt_metal::BufferType::DRAM ? 1 : 0;
+    std::vector<uint32_t> writer_compile_time_args = {
+        (std::uint32_t)output_cb_index, (std::uint32_t)dst_is_dram, (std::uint32_t)num_rows_per_core, (std::uint32_t)num_sin_cos_rows_per_core, Wt, Ht};
+
+    tt_metal::KernelHandle unary_reader_kernel_id = tt_metal::CreateKernel(
+        program,
+        "tt_eager/tt_dnn/op_library/rotary_embedding/kernels/dataflow/reader_rotary_embedding_llama_interleaved_start_id.cpp",
+        all_cores,
+        tt_metal::ReaderDataMovementConfig(reader_compile_time_args, kernel_defines));
+
+    tt_metal::KernelHandle unary_writer_kernel_id = tt_metal::CreateKernel(
+        program,
+        "tt_eager/tt_dnn/op_library/rotary_embedding/kernels/dataflow/writer_rotary_embedding_llama_interleaved_start_id.cpp",
+        all_cores,
+        tt_metal::WriterDataMovementConfig(writer_compile_time_args, kernel_defines));
+
+    vector<uint32_t> compute_kernel_args = {
+        (std::uint32_t)input_cb_index,
+        (std::uint32_t)cos_cb_index,
+        (std::uint32_t)sin_cb_index,
+        (std::uint32_t)trans_mat_cb_index,
+        (std::uint32_t)rotated_input_interm_cb_index,
+        (std::uint32_t)cos_interm_cb_index,
+        (std::uint32_t)sin_interm_cb_index,
+        (std::uint32_t)output_cb_index,
+        (std::uint32_t)num_rows_per_core,
+        (std::uint32_t)num_sin_cos_rows_per_core,
+        (std::uint32_t)Wt,
+        };
+
+    auto rotary_embedding_kernel_id = tt_metal::CreateKernel(
+        program,
+        "tt_eager/tt_dnn/op_library/rotary_embedding/kernels/compute/rotary_embedding_llama.cpp",
+        all_cores,
+        tt_metal::ComputeConfig{.math_fidelity=math_fidelity, .fp32_dest_acc_en=fp32_dest_acc_en, .compile_args = compute_kernel_args, .defines = kernel_defines});
+
+    const auto &cores = grid_to_cores(num_cores, num_cores_x, num_cores_y, row_major);
+
+    uint32_t num_cores_per_sin_cos_row = max((uint32_t) 1, (uint32_t)(num_cores / Ht)); // since sin/cos matrices have Ht rows
+    uint32_t core_idx = 0;
+    /*
+        Overall loop iterations: # total cores
+    */
+
+    std::vector< std::vector<uint32_t> > unary_reader_args = {cores.size(), std::vector<uint32_t>(6)}; // 6 is the number of args in the reader kernel
+    std::vector< std::vector<uint32_t> > unary_writer_args = {cores.size(), std::vector<uint32_t>(2)}; // 2 is the number of args in the writer kernel
+
```

Comment:
Set a default vector with args that are static across cores (eg. buffer addresses) and only update non-static args in the for loop.

Reply:
added default args at initalization and only editing the args that are different across cores.